From: Kevin Lampis <klampis@cloud.com>
Subject: kexec: Block other load types when locked down

When locked down, block load types other than the two new EFI types since the
other load types are not compatible lockdown mode. Also block use of
kexec_load_v1 since the new load types are only implemented for kexec_load.

Signed-off-by: Kevin Lampis <klampis@cloud.com>
diff --git a/xen/common/kexec.c b/xen/common/kexec.c
index f1cfe69f36d2..50cb8c78f93f 100644
--- a/xen/common/kexec.c
+++ b/xen/common/kexec.c
@@ -36,6 +36,7 @@
 #ifdef CONFIG_COMPAT
 #include <compat/kexec.h>
 #endif
+#include <xen/lockdown.h>
 
 bool __read_mostly kexecing;
 
@@ -930,6 +931,9 @@ static int kexec_load(XEN_GUEST_HANDLE_PARAM(void) uarg)
     if ( copy_from_guest(&load, uarg, 1) )
         return -EFAULT;
 
+    if ( load.type != KEXEC_TYPE_CRASH_EFI && is_locked_down() )
+        return -EPERM;
+
     if ( load.nr_segments >= KEXEC_SEGMENT_MAX )
         return -EINVAL;
 
